src/server/protocol.ts

```diff
@@ -625,6 +626,29 @@ export interface GetMoveToRefactoringFileSuggestions extends Response {
     };
 }
 
+/**
+ * Request refactorings at a given position post pasting text from some other location.
+ */
+
+export interface GetPasteEditsRequest extends Request {
+    command: CommandTypes.GetPasteEdits;
+    arguments: GetPasteEditsRequestArgs;
+}
+
+export type GetPasteEditsRequestArgs = FileRequestArgs & {
+    copies: {
+        text: string;
+        range?: FileSpan;
+    }[];
+    pastes: TextSpan[];
+};
+export interface GetPasteEditsResponse extends Response {
+    body: PasteEditsAction;
+}
+export interface PasteEditsAction {

```

Comment:
For the future we will likely want a `name` and/or `id` too so users can tell what type of edit is being made. We can leave that out until there are multiple providers, but means that we'll have to add a flag so older clients don't unexpectedly start using the wrong paste edit